bb8/src/inner.rs

```diff
@@ -161,6 +161,20 @@ where
         }
     }
 
+    /// Adds an external connection to the pool if there is capacity for it.
+    pub(crate) fn try_put(&self, conn: M::Connection) -> Result<(), M::Connection> {
+        let mut locked = self.inner.internals.lock();
+        let mut approvals = locked.approvals(&self.inner.statics, 1);
+        match approvals.next() {
```

Comment:
Suggest spelling this as `let Some(approval) = approvals.next() else { return Err(conn); }`.

Reply:
Sounds good!